core/lib/instrument.c

```diff
@@ -2790,6 +2790,20 @@ dr_get_process_id(void)
     return (process_id_t)get_process_id();
 }
 
+DR_API process_id_t
+dr_get_process_id_ext(void *drcontext)
+{
+    dcontext_t *dcontext = (dcontext_t *)drcontext;
+    CLIENT_ASSERT(drcontext != NULL, "dr_get_process_id_ext: drcontext cannot be NULL");
+    CLIENT_ASSERT(drcontext != GLOBAL_DCONTEXT,
+                  "dr_get_process_id_ext: drcontext is invalid");
+#    ifdef UNIX
+    return dcontext->owning_process;
+#    else
+    return dr_get_process_id();
```

Comment:
Alternatively, the `owning_process` field can be added to `dcontext_t` on Windows too.
Not sure whether it's possible for it to be different from `dr_get_process_id()` there.

Reply:
AFAIK it is not possible on Windows: the process vs thread distinction is much stronger.